client/trino-client/src/main/java/okhttp3/internal/tls/LegacyHostnameVerifier.java

```diff
@@ -19,51 +23,646 @@
 import javax.security.auth.x500.X500Principal;
 
 import java.security.cert.Certificate;
+import java.security.cert.CertificateParsingException;
 import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.regex.Pattern;
 
-import static okhttp3.internal.Util.verifyAsIpAddress;
-import static okhttp3.internal.tls.OkHostnameVerifier.allSubjectAltNames;
-
-public class LegacyHostnameVerifier
+/**
+ * A HostnameVerifier that accepts hostnames in both the standard location (subjectAltName)
+ * and the obsolete CNAME location, which deprecated in RFC 2818 (May 2000).
+ * <p>
+ * Source: https://gist.github.com/swankjesse/461a53ee3078bdaa574e3d4e48aeaf07
+ */
+public final class LegacyHostnameVerifier
```

Comment:
Rather than copy this whole class, we can leave the original implementation and simply copy the last version of [`DistinguishedNameParser`](https://raw.githubusercontent.com/square/okhttp/fc7ac8ae0b30d219cbd884b2af458f19c170c5fb/okhttp/src/main/java/okhttp3/internal/tls/DistinguishedNameParser.java) from OkHttp, which I confirmed is identical the one we are already using. This way, in the normal path, we are using the TLS verification from OkHttp, rather than relying on an implementation from a four year old gist (that is neither tested nor updated with any security fixes). Might also need to copy the  `Util.verifyAsIpAddress()` method, but the others should still be there.